### ISSUE TITLE
Dashed root syntax error (3.0.1 with 1.9.3-p194)

### ISSUE THREAD

**Issue by cmeiklejohn** [jimweirich/builder#27]
Rubygems.org currently has a failure on master when running with builder 3.0.1 and 1.9.3-p194 when generating XML output when the node root is supposed to be a name containing a dash.  See below:

``` ruby
➜  rubygems.org git:(master) ✗ rails c
Unable to load configuration from /Users/cmeiklejohn/Repositories/rubygems.org/config/newrelic.yml
Loading development environment (Rails 3.2.8)
irb(main):001:0> { :a => :b } 
=> {:a=>:b}
irb(main):002:0> { :a => :b }.to_xml
=> "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<hash>\n  <a type=\"symbol\">b</a>\n</hash>\n"
irb(main):003:0> { :a => :b }.to_xml(:root => 'dashed-value')
SyntaxError: (eval):1: syntax error, unexpected '-', expecting ';' or '\n'
        def dashed-value(*args, &block)
                   ^
(eval):3: syntax error, unexpected keyword_end, expecting $end
    from /Users/cmeiklejohn/.rbenv/versions/1.9.3-p194/lib/ruby/gems/1.9.1/gems/builder-3.0.1/lib/builder/xmlbase.rb:182:in `instance_eval'
    from /Users/cmeiklejohn/.rbenv/versions/1.9.3-p194/lib/ruby/gems/1.9.1/gems/builder-3.0.1/lib/builder/xmlbase.rb:182:in `cache_method_call'
    from /Users/cmeiklejohn/.rbenv/versions/1.9.3-p194/lib/ruby/gems/1.9.1/gems/builder-3.0.1/lib/builder/xmlbase.rb:85:in `method_missing'
    from /Users/cmeiklejohn/.rbenv/versions/1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.2.8/lib/active_support/core_ext/hash/conversions.rb:82:in `to_xml'
    from (irb):3
    from /Users/cmeiklejohn/.rbenv/versions/1.9.3-p194/lib/ruby/gems/1.9.1/gems/railties-3.2.8/lib/rails/commands/console.rb:47:in `start'
    from /Users/cmeiklejohn/.rbenv/versions/1.9.3-p194/lib/ruby/gems/1.9.1/gems/railties-3.2.8/lib/rails/commands/console.rb:8:in `start'
    from /Users/cmeiklejohn/.rbenv/versions/1.9.3-p194/lib/ruby/gems/1.9.1/gems/railties-3.2.8/lib/rails/commands.rb:41:in `<top (required)>'
    from script/rails:6:in `require'
    from script/rails:6:in `<main>'
```

**Comment by jimweirich** [comment]
Fixed in 3.0.2.

**Comment by cmeiklejohn** [comment]
Thanks!
